src/Core/Migration/V6_4/Migration1670090989AddIndexOrderOrderNumber.php

```diff
@@ -0,0 +1,29 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration\V6_4;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Framework\Migration\MigrationStep;
+
+/**
+ * @package core
+ *
+ * @internal
+ */
+class Migration1670090989AddIndexOrderOrderNumber extends MigrationStep
+{
+    public function getCreationTimestamp(): int
+    {
+        return 1670090989;
+    }
+
+    public function update(Connection $connection): void
+    {
+        $connection->executeStatement('ALTER TABLE `order` ADD KEY `idx.order_number` (`order_number`)');
```

Comment:
The migration needs to be executed twice or more (in case the migration failed and needs to be retried)

therefore you should check if the index already exists and only add it if it does not yet exist.

And you should add a test for that migration under `tests/migration/php/..`, the existing tests should give you a hint on how to do that.
You should at least cover the two cases that 
1. the index does not exist before and the migration will add it
2. the index does already exist and the migration will not fail/error